### PR TITLE
[sonos] Add support for Era 100 and Era 300

### LINKED ISSUE #14802: [Sonos] 'Sonos Era 100' is not yet a supported model
## Expected Behavior
System should support the new Sonos devices without any warnings.

## Current Behavior
System throws the following warning in the log:
`2023-04-14 11:40:06.843 [WARN ] [overy.ZonePlayerDiscoveryParticipant] - 'Sonos Era 100' is not yet a supported model, thing type 'sonos:zoneplayer' is considered as default; please open an issue`

## Possible Solution
Please add the new Sonos devices as supported model.
As far as I know the following models are new:
- Sonos Era 100 and
- Sonos Era 300

## Steps to Reproduce (for Bugs)
Add a new Sonos model to your openHAB setup.

## Your Environment
Version used: openHAB is 3.4.3 in Docker

### PR DESCRIPTION
**Pull request by lolodomo**
Closes #14802

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
Waiting for information about settings available for these models (bass, treble, loudness, speech enhancement, night mode).
Currently channels for bass, treble and loudness are included but no channels for speech enhancement and night mode.

File: bundles/org.openhab.binding.sonos/README.md [hunk 1]
```diff
@@ -58,7 +58,7 @@ The devices support the following channels:
...
 | batterylevel         | Number    | R           | Current battery level                                                                                                                                     | Move, Roam, RoamSL                                              |
 | clearqueue           | Switch    | W           | Suppress all songs from the current queue                                                                                                                 | all                                                             |
-| codec                | String    | R           | Name of codec currently being decoded                                                                                                                     | Arc, Arc SL, PLAYBAR, PLAYBASE, Beam, Amp                       |
+| codec                | String    | R           | Name of codec currently being decoded                                                                                                                     | Arc, ArcSL, PLAYBAR, PLAYBASE, Beam, Amp                        |
```
Comment (jlaur) [inline bundles/org.openhab.binding.sonos/README.md:61]: If the list refers to actual product names, I believe it was correct before:
```suggestion
| codec                | String    | R           | Name of codec currently being decoded                                                                                                                     | Arc, Arc SL, PLAYBAR, PLAYBASE, Beam, Amp                       |
```

See https://support.sonos.com/en-us/products/arc-sl

File: bundles/org.openhab.binding.sonos/README.md [hunk 2]
```diff
@@ -70,14 +70,14 @@ The devices support the following channels:
...
 | currenttransporturi  | String    | R           | URI of the current AV transport                                                                                                                           | all                                                             |
 | favorite             | String    | W           | Play the given favorite entry. The favorite entry has to be predefined in the Sonos Controller app                                                        | all                                                             |
-| heightlevel          | Number    | RW          | Set or get the height level adjustment (value in range -10 / 10)                                                                                          | Arc, Arc SL                                                     |
+| heightlevel          | Number    | RW          | Set or get the height level adjustment (value in range -10 / 10)                                                                                          | Arc, ArcSL                                                      |
```
Comment (jlaur) [inline bundles/org.openhab.binding.sonos/README.md:73]:
```suggestion
| heightlevel          | Number    | RW          | Set or get the height level adjustment (value in range -10 / 10)                                                                                          | Arc, Arc SL                                                     |
```

File: bundles/org.openhab.binding.sonos/README.md [hunk 3]
```diff
@@ -104,12 +104,12 @@ The devices support the following channels:
...
+| surround             | Switch    | RW          | Enable or disable the surround audio                                                                                                                      | Arc, ArcSL                                                      |
+| surroundmusicmode    | String    | RW          | Set or get the surround playback mode for music, either 0 for Ambient or 1 for full                                                                       | Arc, ArcSL                                                      |
+| surroundmusiclevel   | Number    | RW          | Set or get the surround level adjustment for music (value in range -15 / 15)                                                                              | Arc, ArcSL                                                      |
+| surroundtvlevel      | Number    | RW          | Set or get the surround level adjustment for TV (value in range -15 / 15)                                                                                 | Arc, ArcSL                                                      |
```
Comment (jlaur) [inline bundles/org.openhab.binding.sonos/README.md:112]: And here.

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks for updating with latest models! Only a minor comment about the unrelated Arc SL product name changes.

File: bundles/org.openhab.binding.sonos/README.md (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.sonos/README.md:61]: It refers to thing types, look at column title, that is the reason why I changed ArcSL.

**Review by jlaur — APPROVED** [review]
LGTM. Waiting for signal to merge since still labeled "Work in progress". Will it be tested by a happy owner of an Era speaker?

**Comment by lolodomo** [comment]
> Waiting for signal to merge

@jlaur : you can merge, I just got information from @dankr3 for era 100 and it is was I guessed.

**Pull request merged**